exec_manager/dao/job_dao.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""class for job dao"""
+
+import json
+from uuid import UUID, uuid4
+
+from sqlalchemy import create_engine, insert, select, update
+from sqlalchemy.engine import Engine
+
+from exec_manager.dao.db_models import DBJob, metadata
+from exec_manager.exec_profile import ExecProfile
+from exec_manager.exec_profile_type import ExecProfileType
+from exec_manager.job import Job
+from exec_manager.job_status_type import JobStatusType
+from exec_manager.wf_lang_type import WfLangType
+
+DB_ENGINE = create_engine("sqlite+pysqlite://")
+metadata.create_all(DB_ENGINE)
+
+
+def create_job_dao(
+    job_status: JobStatusType,
+    exec_profile: ExecProfile,
+    workflow: dict,
+    inputs: dict,
+    db_engine: Engine = DB_ENGINE,
+) -> UUID:
+    """
+    Inserts a job into the database.
+
+    Parameters
+    ----------
+    job_status: JobStatusType
+        current status of the job; initially it is JobStatusType.NOTSTARTED
+    exec_profile: ExecProfile
+        exec profile of this job
+    workflow
+        the jobs workflow
+    inputs: dict
+        the input parameters of the job
+    engine: engine
+        db engine where the connection will be established (default is sqlite with pysqlite)
+
+    Returns
+    -------
+    UUID
+    """
+    job_id = generate_job_id()
+    job_id_str = str(job_id)
+    job_status_str = job_status.value
+    exec_profile_json = json.dumps(
+        {
+            "exec_profile_type": exec_profile.exec_profile_type.value,
+            "wf_lang": exec_profile.wf_lang.value,
+        }
+    )
```

Comment:
Wenn das ExecProfile auf einem pydantic model beruht kann man einfach schreiben:

```suggestion
    exec_profile_dict = exec_profile.to_dict()
```
Das hier returned ein dict. Also kein JSON. Aber die Conversion zu JSON kann man auch sqlalchemy machen lassen.